tests/NCronJob.Tests/JobOptionBuilderTests.cs

```diff
@@ -80,4 +81,80 @@ public void ShouldCreateMultipleJobsWithoutAnd()
         options[1].EnableSecondPrecision.ShouldBeFalse();
         options[1].Parameter.ShouldBe("bar");
     }
+
+    [Fact]
+    public void ShouldAssignCorrectTimeZoneToJobOptions()
+    {
+        var builder = new JobOptionBuilder();
+        var timeZone = TimeZoneInfo.FindSystemTimeZoneById("Pacific Standard Time");
+        builder.WithCronExpression("* * * * *", timeZoneInfo: timeZone);
+
+        var options = builder.GetJobOptions();
+
+        options.Single().TimeZoneInfo.ShouldBe(timeZone);
+    }
+
+    [Fact]
+    public void ShouldDefaultToUtcIfTimeZoneNotSpecified()
+    {
+        var builder = new JobOptionBuilder();
+        builder.WithCronExpression("* * * * *");
+
+        var options = builder.GetJobOptions();
+
+        options.Single().TimeZoneInfo.ShouldBe(TimeZoneInfo.Utc);
+    }
+
+    [Fact]
+    public void ShouldCalculateNextRunTimeBasedOnTimeZone()
+    {
+        var builder = new JobOptionBuilder();
+        var timeZone = TimeZoneInfo.FindSystemTimeZoneById("Eastern Standard Time");
+        builder.WithCronExpression("0 12 * * *", timeZoneInfo: timeZone);  // Noon every day
+
+        var options = builder.GetJobOptions();
+
+        var baseTime = new DateTime(2024, 1, 1, 11, 0, 0, DateTimeKind.Utc); // This is 6 AM EST on Jan 1, 2024
+        var expectedRunTime = new DateTime(2024, 1, 1, 17, 0, 0, DateTimeKind.Utc); // Noon EST is 5 PM UTC on Jan 1, 2024
+
+        var nextRunTime = CronExpression.Parse(options.Single().CronExpression)
```

Comment:
I am not a big fan of such tests, mainly because the test tests mainly `Cronos`.
1. That binds tests to. `Cronos` as well
2. It doesn't test that our logic in the scheduler does the right thing.

I would probably use our normal test bed and inject the timeprovider into a job that gets executed that can expose the current `DateTimeOffset`.

Reply:
Good point, I'll see what I could do.

Reply:
This is done. The tests are now acting directly on the Job.